src/faultlog/faultlog_main.cpp

```diff
@@ -0,0 +1,218 @@
+        app.add_flag("-c, --createPel", createPel,
+                     "Create faultlog pel if there are guarded/deconfigured "
+                     "records present");
+        app.add_flag("-f, --faultlog", listFaultlog,
+                     "List all fault log records in JSON format");
+
+        CLI11_PARSE(app, argc, argv);
+
+        // guard records with associated error object
+        if (guardWithEid)
+        {
+            (void)GuardWithEidRecords::populate(bus, unresolvedRecords,
+                                                processedEId, faultLogJson);
+        }
+
+        // guard records without any associated error object
+        else if (guardWithoutEid)
+        {
+            (void)GuardWithoutEidRecords::populate(unresolvedRecords,
+                                                   faultLogJson);
+        }
+
+        // guard policy
+        else if (policy)
+        {
+            (void)FaultLogPolicy::populate(bus, faultLogJson);
+        }
+
+        // unresolved pels with deconfig bit set
+        else if (unresolvedPels)
+        {
+            (void)UnresolvedPELs::populate(bus, unresolvedRecords, processedEId,
+                                           faultLogJson);
+        }
+
+        // pdbg targets with deconfig bit set
+        else if (deconfig)
+        {
+            (void)DeconfigRecords::populate(faultLogJson);
+        }
+
+        // create fault log pel if there are service actions pending
+        else if (createPel)
+        {
+            int guardCount =
+                GuardWithEidRecords::getCount(unresolvedRecords, processedEId);
+            int manualGuardCount =
+                GuardWithoutEidRecords::getCount(unresolvedRecords);
+            int unresolvedPelsCount =
+                UnresolvedPELs::getCount(bus, processedEId);
+            int deconfigRecordsCount = DeconfigRecords::getCount();
+
+            if ((guardCount > 0) || (manualGuardCount > 0) ||
+                (unresolvedPelsCount > 0) || (deconfigRecordsCount > 0))
+            {
+                std::unordered_map<std::string, std::string> data;
+                data.emplace("GUARD_WITH_ASSOC_ERROR_COUNT",
+                             std::to_string(guardCount));
+                data.emplace("GUARD_WITH_NO_ASSOC_ERROR_COUNT",
+                             std::to_string(manualGuardCount));
+                data.emplace("UNRESOLVED_PEL_WITH_DECONFIG_BIT_COUNT",
+                             std::to_string(unresolvedPelsCount));
+                data.emplace("DECONFIG_RECORD_COUNT",
+                             std::to_string(deconfigRecordsCount));
+
+                auto method = bus.new_method_call(
+                    "xyz.openbmc_project.Logging",
+                    "/xyz/openbmc_project/logging",
+                    "xyz.openbmc_project.Logging.Create", "Create");
+                method.append("org.open_power.Faultlog.Error.DeconfiguredHW",
+                              Severity::Critical, data);
```

Comment:
The FSP used a Predictive severity for this.  That may be more appropriate as system is still up and running, just with things deconfigured.  For Critical I would think the system would be crashing or quiescing.

Reply:
i thought d-bus critical maps to fsp level predictive, will change,